overlord/confdbstate/confdbstate.go

```diff
@@ -57,21 +62,38 @@ func SetViaView(bag confdb.Databag, view *confdb.View, requests map[string]inter
 }
 
 // GetView returns the view identified by the account, confdb schema and view name.
-func GetView(st *state.State, account, dbSchemaName, viewName string) (*confdb.View, error) {
-	confdbSchemaAs, err := assertstateConfdbSchema(st, account, dbSchemaName)
+func GetView(st *state.State, account, schemaName, viewName string) (*confdb.View, error) {
+	confdbSchemaAs, err := assertstateConfdbSchema(st, account, schemaName)
 	if err != nil {
-		if errors.Is(err, &asserts.NotFoundError{}) {
-			// replace the not found error so the output matches the usual confdb ID layout
-			return nil, confdb.NewNotFoundError(i18n.G("cannot find confdb-schema %s/%s: assertion not found"), account, dbSchemaName)
+		if !errors.Is(err, &asserts.NotFoundError{}) {
+			return nil, err
+		}
+		logger.Noticef("confdb-schema %s/%s not found locally, fetching from store", account, schemaName)
+
+		userID := 0
+		fetchErr := assertstateFetchConfdbSchemaAssertion(st, userID, account, schemaName)
+		if fetchErr != nil {
+			if errors.Is(fetchErr, store.ErrStoreOffline) {
```

Comment:
can we make a realistic test for this?

Reply:
I can mock the fetching to return ErrStoreOffline and cover this case. The happy path case also mocks it to switch databases so it would be as realistic as the other one

Reply:
Is that an issue? `errors.Is` would unwrap the error chain

Reply:
ah right, that's true. I browsed the assertion code and it looks like we never wrap errors in that call chain (with %w or otherwise). I also added a small test to assertstate to verify it